integration/spark/src/main/scala/org/apache/spark/sql/secondaryindex/rdd/SecondaryIndexCreator.scala

```diff
@@ -385,7 +392,26 @@ object SecondaryIndexCreator {
         val rebuiltSegments = SecondaryIndexUtil
           .mergeDataFilesSISegments(secondaryIndexModel.segmentIdToLoadStartTimeMapping,
             indexCarbonTable,
-            loadMetadataDetails.toList.asJava, carbonLoadModelForMergeDataFiles)(sc)
+            loadMetadataDetail.toList.asJava, carbonLoadModelForMergeDataFiles)(sc)
+        if (isInsertOverwrite) {
+          var segmentList = new ListBuffer[String]()
```

Comment:
rename to more meaningful like what this list does, or required?

Reply:
done